### ISSUE TITLE
Adding trendlines to bar and XY charts

### ISSUE THREAD

**Issue by meacod** [knowm/XChart#478]
Hello, 

Thanks for the great library, it's very useful!

One thing I've been scratching around for, but haven't seemed to find, is the ability to add a trendline through the graph that you render. Is this something you'd plan to add in? I'd find it very useful for showing trends on both line and bar charts.

**Comment by Mr14huashao** [comment]
Is it like this:
![Screenshot_2020-06-24_09-56-26](https://user-images.githubusercontent.com/57353473/85488634-3940ec00-b601-11ea-9088-12182ba82206.png)

You can refer to [BarChart09](https://github.com/knowm/XChart/blob/develop/xchart-demo/src/main/java/org/knowm/xchart/demo/charts/bar/BarChart09.java).

**Comment by meacod** [comment]
Not quite what I was getting at; consider something like the [trendline feature in MS Excel](https://www.dummies.com/wp-content/uploads/excel-data-analysis-logarithmic-trend-line.png) / other alternatives are available.

I would imagine it might be a method call on say a Series or a ChartStyler to add a trendline / line of best fit on top of the provided data.

**Comment by Mr14huashao** [comment]
Add a series, draw line and bar?

**Comment by timmolter** [comment]
No, XChart doesn't have that yet. You'd need to generate the trendline yourself and add a series. I'll leave this open as a feature request.
